neuro_extras/main.py

```diff
@@ -355,99 +363,48 @@ async def _data_cp(
-                raise click.ClickException(f"Extraction failed: {proc.returncode}")
-            else:
-                if file.exists():
-                    # gunzip removes src after extraction, while tar - not
-                    file.unlink()
-
+            # download data into subfolder of target for extraction
+            dst_name = origin_dst_name + "/" + origin_src_url.name
         if compress:
-            file = Path(destination_url.path.split("/")[-1])
-            tmp_dst_archive = Path(TEMP_UNPACK_DIR / file)
-            suffixes = file.suffixes
-            if suffixes[-2:] == [".tar", ".gz"] or suffixes[-1] == ".tgz":
-                command = "tar"
-                args = ["zcf", str(tmp_dst_archive), "-C", str(source_url.path), "."]
-            elif suffixes[-2:] == [".tar", ".bz2"] or suffixes[-1] in (".tbz2", ".tbz"):
-                command = "tar"
-                args = ["jcf", str(tmp_dst_archive), str(source_url.path)]
-            elif suffixes[-1] == ".tar":
-                command = "tar"
-                args = ["cf", str(tmp_dst_archive), str(source_url.path)]
-            elif suffixes[-1] == ".gz":
-                command = "gzip"
-                args = ["-r", str(tmp_dst_archive), str(source_url.path)]
-            elif suffixes[-1] == ".zip":
-                command = "zip"
-                args = [str(tmp_dst_archive), str(source_url.path)]
-            else:
-                raise ValueError(
-                    f"Don't know how to compress to archive type {file.name}. "
-                    f"Supported archive types: {', '.join(SUPPORTED_ARCHIVE_TYPES)}"
+            # preserve origin destination name for compressor
+            if origin_dst_name == origin_src_url.name:
+                logging.warning(
+                    "Source file already has required archive extension. "
+                    "Skipping compression step."
                 )
-
-            click.echo(f"Running {command} {' '.join(args)}")
-            subprocess = await asyncio.create_subprocess_exec(command, *args)
-            returncode = await subprocess.wait()
-            if returncode != 0:
-                raise click.ClickException(f"Compression failed: {subprocess.stderr}")
+                compress = False
             else:
-                source_url = URL(str(tmp_dst_archive))
-                # At this moment we know destination URL is a file, but we need its
-                # parent directory
-                destination_file = Path(destination_url.path)
-                destination_dir = destination_file.parent
-                destination_url = URL.build(
-                    scheme=destination_url.scheme, path=str(destination_dir)
-                )
+                dst_name = origin_src_url.name
 
-        # handle upload/rclone
-        await _nonstorage_cp(source_url, destination_url, remove_source=True)
+        cp_destination_url = URL.build(path=str(dst_dir / dst_name))
```

Comment:
oh we do need to refactor this method, it's hard to understand

Reply:
I would be cool, but the only way I see it - apply some oop pattern as we discussed in the early beginning.